pkg/compare/compare.go

```diff
@@ -450,9 +495,14 @@ func (o *Options) Run() error {
 			return err
 		}
 
+		localRef, patch, err := mergeAndGetPatch(localRef, clusterCR, o.showMore)
```

Comment:
There is a dedicated section in the code for the merging part : https://github.com/openshift/kube-compare/blob/main/pkg/compare/compare.go#L499-L503
Its better to move it there, its pare of the diff interface  

just as an example  you can look a[t he implementation of the function in kubectl diff,](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/diff/diff.go#L346) they do a different merge but it shows the general idea